### PR TITLE
cherry pick modem subsystem

### PR DESCRIPTION
**Pull request by tomi-font**

### CODE REVIEW

**Review by SeppoTakalo — APPROVED** [review]

File: drivers/gnss/Kconfig.generic [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Copyright 2023 Google LLC
```
Comment (MarkusLassila) [inline drivers/gnss/Kconfig.generic:1]: Are these GNSS changes somehow intertwined with the modem driver changes? I was not expecting to see these in here.

Comment (tomi-font) [reply drivers/gnss/Kconfig.generic:1]: Yeah they are to some extent as I encountered some merge conflicts without taking in those GNSS changes.

Comment (MarkusLassila) [reply drivers/gnss/Kconfig.generic:1]: I think that the modem driver stuff is likely OK, as we are the only user for many of those, however, I am a bit worried about these GNSS changes. Ping @trantanen.

**Review by MarkusLassila — APPROVED** [review]
OK from SLM perspective. We need the modem driver changes.

File: drivers/gnss/Kconfig.generic (hunk 1 above)
Comment (MarkusLassila) [reply drivers/gnss/Kconfig.generic:1]: Sorry, wrong Tommi. Ping @tokangas.

Comment (tokangas) [reply drivers/gnss/Kconfig.generic:1]: The Zephyr GNSS API is already present in sdk-zephyr, but we don't use it currently. So, these commits should be fine.

**Review by rlubos — APPROVED** [review]

**Pull request merged**